### PR TITLE
Upgrade to enzyme 3

### PR DESCRIPTION
**Pull request by cjwatson**
This will make upgrading React somewhat easier in future.

See: https://airbnb.io/enzyme/docs/guides/migration-from-2-to-3.html

### CODE REVIEW

File: test/unit/src/common/components/repositories-home/t_repositories-home.js [hunk 1]
```diff
@@ -39,7 +39,9 @@ describe('The RepositoriesHome component', () => {
         router: {}
       };
 
-      wrapper = shallow(<RepositoriesHome { ...props } />);
+      wrapper = shallow(
+        <RepositoriesHome { ...props } />, { disableLifecycleMethods: true }
```
Comment (bartaz) [inline test/unit/src/common/components/repositories-home/t_repositories-home.js]: You are only disabling it for tests that were failing with this enabled?

Comment (cjwatson) [reply test/unit/src/common/components/repositories-home/t_repositories-home.js]: Correct.  (Would you prefer I disabled it globally?  I don't have a strong opinion.)

Comment (bartaz) [reply test/unit/src/common/components/repositories-home/t_repositories-home.js]: On one hand turning it off globally would keep it consistent with version we've been using, but if the change affected only couple of tests (as it seems), I guess it's fine to stick with new default and only update where needed. So new tests can be written with default enzyme behaviour from the start.

So I'm fine with keeping it as it is. Thanks.

Comment (cjwatson) [reply test/unit/src/common/components/repositories-home/t_repositories-home.js]: Thanks for articulating something equivalent to my rather vague intuition :-)

**Pull request merged**